### PR TITLE
JupyterLab+Colab extension

### PR DESCRIPTION
**Pull request by DanielSchiavini**
Fixes https://github.com/vyperlang/titanoboa/issues/83

- Fix pre-commit checks
- Create a jupyter server extension
- Inject ethers and boa code via Python Javascript
- Implement front-end callbacks via JS file
- Simplify the browser signer
- Usage example available in https://github.com/vyperlang/try.vyperlang.org/pull/1
- Add support for Google Colab
- [ ] Figure out how to implement tests with jupyter dependencies

### Usage
```python
import boa
from boa.integrations.jupyter import BrowserSigner
from boa.network import NetworkEnv

boa.set_env(NetworkEnv(...))
boa.env.add_account(BrowserSigner())
```

## Cute animal in Jupiter

image
[source](https://steemit.com/space/@danielwooddesign/steemians-view-my-pets-in-space-jupiter-artwork-by-danielwooddesign)

### CODE REVIEW

**Comment by DanielSchiavini** [comment]
This PR was drafted because we wanted to investigate support for notebook v6.

Since v6 is [deprecated](https://github.com/jupyter/notebook?tab=readme-ov-file#classic-notebook-v6) and works quite [differently](https://stackoverflow.com/a/75515387/977026), I discussed with @charles-cooper that it was best to support JupyterLab and Google Colab only.

An alternative solution for Google Colab is provided in a follow-up PR.

**Comment by DanielSchiavini** [comment]
https://github.com/vyperlang/titanoboa/blob/2c2a2664ad3fa566ddda8a43851cd6f14ab9276a/boa/integrations/jupyter.py#L101

File: boa/integrations/jupyter/signer.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    :param snippet: A function that given a token and some kwargs, returns a Javascript snippet.
+    :param kwargs: The arguments to pass to the Javascript snippet.
+    :return: The result of the Javascript snippet sent to the API.
+    """
+    install_jupyter_javascript_triggers()
+
+    token = _generate_token()
+    javascript = snippet(token, **kwargs)
+
+    if importlib.util.find_spec("google.colab"):
```
Comment (charles-cooper) [inline boa/integrations/jupyter/signer.py:75]: dark magic :)

**Review by charles-cooper — APPROVED** [review]
incredible work. thank you!!!

**Pull request merged**